src/bridge/cockpitrouter.c

```diff
@@ -63,6 +64,121 @@ enum {
 };
 
 typedef struct {
+  JsonObject *config;
+
+  // Contents owned by config
+  gchar **env;
+  gchar **spawn;
+
+  GHashTable *peers;
+
+} DynamicPeer;
+
+/* TODO: GLIB JSON 1.2 has better hash functions
+ * once we are able to bump our dependencies
+ * we should use them instead
+ */
+static guint
+string_array_hash (gconstpointer key)
+{
+  JsonArray *array;
+  guint hash = 0;
+  guint i;
+  guint length;
+
+  g_return_val_if_fail (key != NULL, 0);
+
+  array = (JsonArray *) key;
+
+  length = json_array_get_length (array);
+  for (i = 0; i < length; i++)
+    {
+      JsonNode *node = json_array_get_element (array, i);
+      if (node && JSON_NODE_HOLDS_VALUE (node) && json_node_get_value_type (node) == G_TYPE_STRING)
+        hash ^= (i ^ g_str_hash (json_node_get_string (node)));
+      else
+        g_warning ("Error can't hash non-string elements in json arrays");
```

Comment:
```g_return_if_reached ()``` is customary in such a situation. Where the input to this function is clearly invalid.